worker/uniter/actions/resolver.go

```diff
@@ -42,13 +42,13 @@ func (r *actionsResolver) NextOp(
 	opFactory operation.Factory,
 ) (operation.Operation, error) {
 	nextAction, err := nextAction(remoteState.Actions, localState.CompletedActions)
-	if err != nil {
+	if err != nil && err != resolver.ErrNoOperation {
 		return nil, err
 	}
 	switch localState.Kind {
 	case operation.RunHook:
 		// We can still run actions if the unit is in a hook error state.
-		if localState.Step == operation.Pending {
+		if localState.Step == operation.Pending && err != resolver.ErrNoOperation {
```

Comment:
If we are here, then either err == nil or err == ErrNoOperation
I think it would read much nicer to say
if localState.Step == operation.Pending && err == nil